src/python/pants/build_graph/build_configuration.py

```diff
@@ -30,24 +31,34 @@
 
 
 # Subsystems used outside of any rule.
-_GLOBAL_SUBSYSTEMS: FrozenOrderedSet[Type[Subsystem]] = FrozenOrderedSet({GlobalOptions, Changed})
+_GLOBAL_SUBSYSTEMS: Set[Type[Subsystem]] = {GlobalOptions, Changed}
 
 
 @dataclass(frozen=True)
 class BuildConfiguration:
     """Stores the types and helper functions exposed to BUILD files."""
 
     registered_aliases: BuildFileAliases
-    subsystems: FrozenOrderedSet[Type[Subsystem]]
+    subsystem_to_providers: FrozenDict[Type[Subsystem], Tuple[str, ...]]
```

Comment:
AFAICT the ordering of subsystems and target_types here was never important or significant, but please correct me if I'm wrong about that.

It's not clear to me that the rules/union_rules need to be ordered either. Or at least, the current order is registration order, which seems not significant and subject to change. If we need rules in a consistent order (perhaps to ensure consistent graph construction?) then maybe that should be alphabetical order of rule name or order of rule signature? Still, this is all future musing, and not for this change.

Reply:
`union_rules` ordering matters currently: their registration order is preserved, and ends up being their order in the `UnionMembership`, which drives what order formatters are applied in, which could hypothetically matter. I think that's the only place currently.

`rule` ordering shouldn't matter as long as it is stable with a given set of backends (to avoid error messages changing their rendering, etc).

Reply:
Where order matters, we might want to determine an order more meaningful than "registration order", which is fairly arbitrary. Nowhere as far as I can tell do we ensure that registration happens in a certain useful order.